src/unity/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -145,19 +146,49 @@ void activity_classifier::init_options(
   add_or_update_state(flexmap_to_varmap(options.current_option_values()));
 }
 
+std::tuple<float, float> activity_classifier::compute_validation_metrics(
+    data_iterator *validation_data_iterator_, size_t prediction_window,
```

Comment:
I don't think we need to pass the `data_iterator*` since we can just use the class data member, right?